Makefile

```
@@ -201,8 +207,9 @@ release: $(GO)
 	hack/release.sh
 
 vendor: $(GO)
-	$(GO) mod tidy
-	$(GO) mod vendor
+	for f in `find . -name go.mod -not -path '*/vendor/*' -not -path '*/build/*'`; do \
+		cd $$(dirname $$f); $(GO) mod tidy; $(GO) mod vendor; cd $(PWD); \
+	done
```

We are not going to have more sub modules, let's just create a target for api.

```suggestion
vendor: vendor-api                                                              
    $(GO) mod tidy                                                              
    $(GO) mod vendor                                                            
                                                                                
vendor-api: $(GO)                                                               
    cd api && $(GO) mod tidy && $(GO) mod tidy                
```

---

There are two submodles that i had to create, `api` and  `pkg/names`. 
`pkg/namese` are used by api so unless it's a submodule it will create a cyclic dependency. 
I can create multiple targets, just was thinking about something more generic.

---

fixed